### ISSUE TITLE
WirePattern fires multiple times from addressable inport

### ISSUE THREAD

**Issue by ComFreek** [noflo/noflo#273]
Component repository: https://github.com/ComFreek/noflo-test
JS file: https://github.com/ComFreek/noflo-test/blob/master/components/test.js

The network:
(The two Core/Repeat nodes have been set to `5` and `10` respectively.)
![Graph, two Core/Repeat Inputs plugged into one input from the above component](https://cloud.githubusercontent.com/assets/1827709/6105775/2ac283a2-b05b-11e4-99d9-9e595f6e88d7.PNG)

The output when the graph is run:
```
New data from 'WirePattern'
{ '0': '5' }
New data from 'WirePattern'
{ '1': '10' }
```
Am I missing some important aspect or why doesn't WirePattern fire only once with both objects being passed as parameters to the callback?
My actual use case: The component should wait for all inputs plugged into it and then process the data it received from all those components in one go.

**Comment by trustmaster** [comment]
Confirmed, the bug affects WirePattern if there is only 1 input port. The specs use at least 2 input ports and they pass.

**Comment by marclaporte** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  marc@laporte.name
    Domain marclaporte.com has exceeded the max emails per hour (313/300 (104%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o3.sgmail.github.com ([192.254.112.98]:2448)
	by uskm12.fastbighost.net with esmtps (TLSv1.2:DHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.84)
	(envelope-from <bounces+848413-0bd8-marc=marclaporte.com@sgmail.github.com>)
	id 1YKref-002VO4-Cw
	for marc@marclaporte.com; Mon, 09 Feb 2015 18:57:50 +0200
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed; d=github.com; 
	h=from:reply-to:to:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=aW2RuOM5TS/5oF4gAHjT/UNyf64=; b=iFLO8mrqe4Btk1e9
	flPEo4Cgo8YLhUVYJ2UwNmCoWgjkileensuwcyIHCRZwfgaNDYYmQFe2OstZZ3XN
	CEGBFewGhQNYSA3AsnkZP3JowOQG7kqFGV7qDWC8segcfcOVsiWzV4Dv6U/VsxQC
	/RIO5mF/ie0G9AdBUadssV5HRl4=
Received: by filter0044p1mdw1.sendgrid.net with SMTP id filter0044p1mdw1.14209.54D8E70216
        2015-02-09 16:57:40.286650681 +0000 UTC
Received: from github-smtp2b-ext-cp1-prd.iad.github.net (github-smtp2b-ext-cp1-prd.iad.github.net [192.30.253.17])
	by ismtpd-019 (SG) with ESMTP id 14b6f466885.5608.ee895
	for <marc@marclaporte.com>; Mon, 09 Feb 2015 16:57:40 +0000 (UTC)
Date: Mon, 09 Feb 2015 08:57:40 -0800

**Comment by bergie** [comment]
Merged the fix in #274

**Comment by ComFreek** [comment]
Thanks, but NoFlo now throws an error (using the same component as mentioned above):

```
Error: No process defined for inbound node test/Test_a04e8
```

The error is apparently very serious because I can't run the graph after the error appeared. I must then completely restart the `noflo-nodejs` process.

**Comment by trustmaster** [comment]
Any error output in noflo-nodejs console? Is it using the current git master of noflo?

**Comment by ComFreek** [comment]
@trustmaster There is no output. I actually tried setting the dependencies to their respective Git URI from GitHub, but then I could not get `noflo-nodejs` (`node node_modules/noflo-nodejs/bin/noflo-nodejs`) to work because it didn't find the `noflo` package. I could resolve the issue by replacing the Git URIs by normal version numbers. There must be either a serious error in NoFlo or in Node.js.

**Comment by trustmaster** [comment]
I just wonder how you got the fix applied to your runtime so that it started throwing that error. Git repo urls don't always work. The quick and dirty trick to have an unreleased noflo dependency running is to clone it to a separate repo, run `npm install` in there and then just copy the files over to `node_modules`, replacing an existing package there.

**Comment by ComFreek** [comment]
I've now done that with this repository and copied the files to `node_modules` (even needed to use 7-zip file manager since Windows cannot handle too long filenames) as you said.
I'm still getting the `No process defined ...`error though.

**Comment by trustmaster** [comment]
Ok, i'll check it later this week.

**Comment by trustmaster** [comment]
Sorry @ComFreek, couldn't get it working so far. Currently getting

> Registration with Flowhub failed: CERT_HAS_EXPIRED

error when trying to run a noflo-nodejs runtime. Am I the only one who gets this?

**Comment by ComFreek** [comment]
@trustmaster Yes, I am getting the same error. I can't even re-login to noflo-ui. I am always being redirected to https://passport.thegrid.io/login/github_public/callback?code=xyz whose output is:
```
Error: Identity connected to another user
 at /app/auth.coffee:52:27
 at /app/lib.coffee:92:5
 at tryCatch1 (/app/node_modules/knex/node_modules/bluebird/js/main/util.js:45:21)
 at Promise$_callHandler [as _callHandler] (/app/node_modules/knex/node_modules/bluebird/js/main/promise.js:660:13)
 at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/app/node_modules/knex/node_modules/bluebird/js/main/promise.js:675:18)
 at Promise$_settlePromiseAt [as _settlePromiseAt] (/app/node_modules/knex/node_modules/bluebird/js/main/promise.js:845:14)
 at Promise$_settlePromises [as _settlePromises] (/app/node_modules/knex/node_modules/bluebird/js/main/promise.js:988:14)
 at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/app/node_modules/knex/node_modules/bluebird/js/main/async.js:77:12)
 at Async$consumeFunctionBuffer (/app/node_modules/knex/node_modules/bluebird/js/main/async.js:40:14)
 at /app/node_modules/newrelic/node_modules/continuation-local-storage/node_modules/async-listener/glue.js:188:31
 at process._tickCallback (node.js:442:13)
```

**Comment by trustmaster** [comment]
Seems like we need some help from @bergie here.

**Comment by forresto** [comment]
@ComFreek issue should be fixed now.

**Comment by ComFreek** [comment]
@forresto (Y) Thanks for Fixing It works fine now (except for this issue of course).

**Comment by trustmaster** [comment]
@ComFreek I've checked your example. In Flowhub it was the same for me as in your 1st post because it kept using browser runtime of NoFlo no matter what i tried.

Then I ran it in console. First with a published 0.5.11 noflo build:

```
$ ./node_modules/.bin/noflo graphs/TestGraph.json
New data from 'WirePattern'
{ '0': '5' }
New data from 'WirePattern'
{ '1': '10' }
```

Then with a patched noflo from git:

```
$ ./node_modules/.bin/noflo graphs/TestGraph.json
New data from 'WirePattern'
{ '0': '5', '1': '10' }
```

Here is the gist: https://gist.github.com/trustmaster/b49965fcd4499964ecdf

**Comment by ComFreek** [comment]
@trustmaster 
Which packages did you update?
I have already updated `noflo` and `noflo-nodejs` to no avail. I am still getting the same error.

**Comment by trustmaster** [comment]
Don't forget that you need `noflo-core` on Node.js.

**Comment by ComFreek** [comment]
I have now updated `noflo-core` as well and I am still getting the same error.
I also verified that my custom component is based on the same code as yours in the Gist.

Do I have to update noflo-ui maybe?
